python/ray/data/preprocessors/tensorizer.py

```diff
@@ -0,0 +1,73 @@
+from typing import List, Optional
+import numpy as np
+import pandas as pd
+
+from ray.data.extensions import TensorArray
+from ray.data.preprocessor import Preprocessor
+
+
+class Tensorizer(Preprocessor):
+    """Create tensor columns via concatenation.
+
+    A tensor column is a a column consisting of ndarrays as elements.
+
+    Example:
+        >>> import pandas as pd
+        >>> from ray.data.preprocessors import Tensorizer
+        >>> df = pd.DataFrame({"a": [1, 2, 3, 4], "b": [1, 2, 3, 4],})
+        >>> ds = ray.data.from_pandas(df)
+        >>> prep = Tensorizer(["a", "b"], "c")
+        >>> new_ds = prep.transform(ds)
+        >>> df = new_ds.to_pandas()
+        #         c
+        # 0  [1, 1]
+        # 1  [2, 2]
+        #       ...
+        >>> x = df["c"].iloc[0]
+        >>> assert x.to_numpy().tolist() == [1, 1]
+
+    Args:
+        columns: A list of column names that should be
+            concatenated into a single column. After concatenation,
+            these columns will be dropped.
+        output_column: output_column is a string that represents the
+            name of the outputted, concatenated tensor.
+        dtype: Optional. The dtype to convert the output column array to.
+
+    Raises:
+        ValueError if any element in `columns` does not exist in the dataset.
+    """
+
+    _is_fittable = False
+
+    def __init__(
+        self,
+        columns: List[str],
+        output_column: str,
```

Comment:
Shall we set a default of "__value__" here to be consistent with single-column datasets?

Reply:
This `VALUE_COL_NAME` isn't publicly documented anywhere right? Seems like we should first at least document it and move it out of `_internal` before using it?

Reply:
We should definitely document this (cc @clarkzinzow ). Btw, we can use it as a default arg without moving it out of internal, that's totally fine.

Reply:
Actually, maybe better to use something like concat_out given we preserve other columns.

Reply:
OK, that's fine for the default. Will change. 

For the getting started example though, we shouldn't use this default value.

The trainer code later requires knowing what the keys are in order to index the iter_batches output properly, and having a __value__ magic keyword is very unintuitive.

Reply:
saw your comment below, sounds good.